### PR TITLE
port main to kotlin

### PR DESCRIPTION
**Pull request by bbaldino**
I've been doing more work in main and wishing it was in Kotlin.  It also makes a lot of the null-check code cleaner.

### CODE REVIEW

File: jvb/src/main/kotlin/org/jitsi/videobridge/Main.kt [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    logger.info("Bridge shutting down")
+    octoRelayService?.stop()
+    clientConnection.stop()
+    statsMgr?.stop()
+
+    try {
+        publicHttpServer?.stop()
+        privateHttpServer?.stop()
+    } catch (t: Throwable) {
+        t.printStackTrace()
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/Main.kt:114]: Print through the logger now that we have one?

File: jvb/src/main/kotlin/org/jitsi/videobridge/Main.kt [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+private fun setupPrivateHttpServer(): Server? {
+    val privateServerConfig = JettyBundleActivatorConfig(
+        "org.jitsi.videobridge.rest.private",
+        "videobridge.http-servers.private"
+    )
+    if (privateServerConfig.port == -1 && privateServerConfig.tlsPort == -1) {
+        return null
+    }
+    val privateServer = createServer(privateServerConfig)
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/Main.kt:163]: `return createServer(privateServerConfig).apply { ... }`?

File: jvb/src/main/kotlin/org/jitsi/videobridge/Main.kt [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        "org.jitsi.videobridge.rest",
+        "videobridge.http-servers.public"
+    )
+    if (publicServerConfig.port == -1 && publicServerConfig.tlsPort == -1) {
+        return null
+    }
+    val publicServer = createServer(publicServerConfig)
+
+    val websocketService = ColibriWebSocketService(publicServerConfig.isTls)
+    org.jitsi.videobridge.websocket.singleton().setColibriWebSocketService(websocketService)
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/Main.kt:148]: You could import all these with shorter names, e.g. `websocketSingleton`

**Review by bgrozev — APPROVED** [review]

File: jvb/src/main/kotlin/org/jitsi/videobridge/Main.kt (hunk 3 above)
Comment (bbaldino) [reply jvb/src/main/kotlin/org/jitsi/videobridge/Main.kt:148]: Yeah, I can do it that way.

**Review by bgrozev — APPROVED** [review]

**Pull request merged**